### ISSUE TITLE
Does not report undefined variable when overriding property descriptor in a subclass

### ISSUE THREAD

**Issue by mattwala** [PyCQA/pylint#3641]
### Steps to reproduce

```python
class A:
    @property
    def x(self):
        raise NotImplementedError

class B(A):
    @x.getter
    def x(self):
        return 42
```

### Current behavior

Pylint output suggests this code has no errors:
```
$ pylint -E test.py
$ # no errors
```

### Expected behavior

The name `x` is not in scope (the correct way to override this would be `@A.x.getter`).

Python output:
```
$ python test.py 
Traceback (most recent call last):
  File "test.py", line 8, in <module>
    class B(A):
  File "test.py", line 10, in B
    @x.getter
NameError: name 'x' is not defined
```

### pylint --version output

```
pylint 2.5.2
astroid 2.4.1
Python 3.8.2 | packaged by conda-forge | (default, Mar 23 2020, 17:55:48) 
[Clang 9.0.1 ]
```

**Comment by mattwala** [comment]
This issue is more general than properties. Pylint doesn't report an error for this code:

```python
class C:
    @f
    def f(self):
        pass
```
